lock-api-objects/src/main/java/com/palantir/lock/v2/AcquireNamedMinTimestampLeaseResult.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import com.palantir.timestamp.TimestampRange;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface AcquireNamedMinTimestampLeaseResult {
+    @Value.Parameter
+    long getMinLeased();
+
+    @Value.Parameter
+    LockToken getLock();
```

Comment:
Would it make sense to make sure this is a LeasedLockToken?

Once you start implementing these, you'll realize that LockLeaseService does not work with LockTokens, everything needs to be LeasedLockToken. It would be great to just start enforcing this, instead of this being a silly low level piece of knowledge you need to know about.